BeeSwift/Gallery/GalleryViewController.swift

```diff
@@ -340,46 +348,53 @@ class GalleryViewController: UIViewController, UICollectionViewDelegateFlowLayou
     }
 
     func updateGoals() {
-        let goals = goalManager.staleGoals(context: viewContext) ?? []
-        self.goals = sortedGoals(goals)
         self.updateFilteredGoals()
         self.didUpdateGoals()
     }
-
-    func sortedGoals(_ goals: any Sequence<Goal>) -> [Goal] {
+    
+    private var preferredSort: [NSSortDescriptor] {
         let selectedGoalSort = UserDefaults.standard.value(forKey: Constants.selectedGoalSortKey) as? String ?? Constants.urgencyGoalSortString
         
         switch selectedGoalSort {
         case Constants.nameGoalSortString:
-            return goals.sorted(using: [SortDescriptor(\.slug),
-                                        SortDescriptor(\.urgencyKey)])
+            return [
+                NSSortDescriptor(keyPath: \Goal.slug, ascending: true),
+                NSSortDescriptor(keyPath: \Goal.urgencyKey, ascending: true)
+            ]
         case Constants.recentDataGoalSortString:
-            return goals.sorted(using: [SortDescriptor(\.lastTouch, order: .reverse),
-                                        SortDescriptor(\.urgencyKey)])
+            return [
+                NSSortDescriptor(keyPath: \Goal.lastTouch, ascending: true),
+                NSSortDescriptor(keyPath: \Goal.urgencyKey, ascending: true)
+            ]
         case Constants.pledgeGoalSortString:
-            return goals.sorted(using: [SortDescriptor(\.pledge, order: .reverse),
-                                        SortDescriptor(\.urgencyKey)])
+            return [
+                NSSortDescriptor(keyPath: \Goal.pledge, ascending: true),
+                NSSortDescriptor(keyPath: \Goal.urgencyKey, ascending: true)
+            ]
         default:
-            return goals.sorted(using: SortDescriptor(\.urgencyKey))
+            return [
+                NSSortDescriptor(keyPath: \Goal.urgencyKey, ascending: true)
+            ]
         }
     }
-
+    
     func updateFilteredGoals() {
-        let searchText = searchBar.text ?? ""
-
-        if searchText == "" {
-            self.filteredGoals = self.goals
+        if let searchText = searchBar.text, !searchText.isEmpty {
+            self.fetchedResultsController.fetchRequest.predicate = NSPredicate(format: "slug contains[cd] %@", searchText)
         } else {
-            self.filteredGoals = self.goals.filter { (goal) -> Bool in
-                return goal.slug.lowercased().contains(searchText.lowercased())
-            }
+            self.fetchedResultsController.fetchRequest.predicate = nil
         }
+        
+        self.fetchedResultsController.fetchRequest.sortDescriptors = preferredSort
+        try? self.fetchedResultsController.performFetch()
+        
+        self.applySnapshot()
```

Comment:
Do we need to `applySnapshot()` here, or will the delegate callback method handle that?